### ISSUE TITLE
zlib error: unexpected end of file

### ISSUE THREAD

**Issue by iqqmuT** [omni-resources/civ6-save-parser#7]
When using `--outputCompressed` option, zlib throws following error:

```
zlib.js:489
      throw error;
      ^

Error: unexpected end of file
    at Zlib.zlibOnError [as onerror] (zlib.js:153:17)
    at processChunkSync (zlib.js:481:12)
    at zlibBufferSync (zlib.js:139:12)
    at Object.unzipSync (zlib.js:695:14)
    at readCompressedData (/tmp/foop/index.js:472:33)
    at Object.module.exports.parse (/tmp/foop/index.js:113:9)
    at Object.<anonymous> (/tmp/foop/index.js:217:35)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
```

**Comment by mrosack** [comment]
Thanks for the attempt at the PR, @iqqmuT, I swear this worked at some point in time but I can't figure out what the problem is now.  I'll play around with it as I have time.

**Comment by iqqmuT** [comment]
It seems to work with some save files in `test/saves/` directory, but not for all.

✔️ This works:
```
$ node index.js test/saves/000002.Civ6Save --outputCompressed
```
✖️ But not this, gives another error `invalid code lengths set`:
```
$ node index.js test/saves/000144.Civ6Save --outputCompressed
zlib.js:489
      throw error;
      ^

Error: invalid code lengths set
    at Zlib.zlibOnError [as onerror] (zlib.js:153:17)
    at processChunkSync (zlib.js:481:12)
    at zlibBufferSync (zlib.js:139:12)
    at Object.unzipSync (zlib.js:695:14)
    at readCompressedData (/home/tumppi/projects/civ6-save-parser/index.js:470:33)
    at Object.module.exports.parse (/home/tumppi/projects/civ6-save-parser/index.js:113:9)
    at Object.<anonymous> (/home/tumppi/projects/civ6-save-parser/index.js:217:35)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
```

I managed to see also this error with my another save:

```
zlib.js:489
      throw error;
      ^

Error: invalid literal/length code
    at Zlib.zlibOnError [as onerror] (zlib.js:153:17)
    at processChunkSync (zlib.js:481:12)
    at zlibBufferSync (zlib.js:139:12)
    at Object.unzipSync (zlib.js:695:14)
    at readCompressedData (/home/tumppi/projects/civ6-save-parser/index.js:470:33)
    at Object.module.exports.parse (/home/tumppi/projects/civ6-save-parser/index.js:113:9)
    at Object.<anonymous> (/home/tumppi/projects/civ6-save-parser/index.js:217:35)
    at Module._compile (internal/modules/cjs/loader.js:702:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:713:10)
    at Module.load (internal/modules/cjs/loader.js:612:32)
```

I'm using node v10.3.0.

**Comment by iqqmuT** [comment]
Sorry, it does not work with any of save files. I forgot to remove `{ finishFlush: zlib.Z_SYNC_FLUSH }` from my tests.

**Comment by mrosack** [comment]
I got it to work using streams to decompress the data instead of unzipSync, although I have no idea why it works this way and not the other.  Let me know if you have any thoughts on what's in the compressed data, thanks!

fixed in 20520fbd28bb63dc11d8bf1517d5bc50151c6545
